### ISSUE TITLE
[Feature Request] Hitomi.la DL enhancement

### ISSUE THREAD

**Issue by Ero-gamer** [avluis/Hentoid#1158]
Feature request to add option to choose between native encoded image format: avif and jxl over webp, jpeg and png for downloads.

When I save images from hitomi.la directly while on browser, I get them as .avif image format which is better compressed compared to .webp. But when I download archive or images from Hentoid app, I always get .webp images, and there is no way to download .avif or .jxl image formats.

Possible solution: make saving individual images from Hitomi.la automated.

**Comment by Ero-gamer** [comment]
> instead of one site how about adding avif options on transform?

That would be great but avif encoding in current state is 10 times slower (or more) than webp encoding, not to mention only android 12+ support it natively and very few apps decode avif although I have some good manga/book reader apps which support it on android 7+.
The reason I asked for this enhancement is to the option to download avif over webp since it's already available on HL website which save me the trouble of encoding.

**Comment by sadboy889** [comment]
how to download jxl? i can download avif like you mentioned but no way for jxl

**Comment by Ero-gamer** [comment]
> how to download jxl? i can download avif like you mentioned but no way for jxl

Not all galleries on HL are encoded in jxl format, only few of them. Try HDoujin downloader on windows, it has image format encoding and various advanced features.

**Comment by sadboy889** [comment]
> > how to download jxl? i can download avif like you mentioned but no way for jxl
> 
> Not all galleries on HL are encoded in jxl format, only few of them. Try HDoujin downloader on windows, it has image format encoding and various advanced features.

hi, its available on mihon now with hitomi extension also i think gallery-dl support this as well

**Comment by Ero-gamer** [comment]
> > > how to download jxl? i can download avif like you mentioned but no way for jxl
> > 
> > 
> > Not all galleries on HL are encoded in jxl format, only few of them. Try HDoujin downloader on windows, it has image format encoding and various advanced features.
> 
> hi, its available on mihon now with hitomi extension also i think gallery-dl support this as well

Yes, I already know that. [EhView Enhance](https://github.com/MapoMagpie/eh-view-enhance) script also added jxl feature recently, you can install it on all desktop internet browsers that support scripts/extensions or install it on android on: Kiwi browser, Firefox android, Xbrowser or VIA browser.

**Comment by RobbWatershed** [comment]
AVIF support has been implemented and will be available for next release.

JXL seems to have been dropped by Hitomi, at least on recent books

**Comment by Ero-gamer** [comment]
> AVIF support has been implemented and will be available for next release.
> 
> JXL seems to have been dropped by Hitomi, at least on recent books

Thank you for your hard work.
